lib/connectors/memory.js

```diff
@@ -679,6 +691,41 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replaceAttributes = function replaceAttributes(model, id, data, options, cb) {
+  if (!id) {
+    var err = new Error('You must provide an id when replacing attributes!');
+    if (cb) {
+      return cb(err);
+    } else {
+      throw err;
+    }
+  }
+  this.setIdValue(model, data, id);
```

Comment:
Please create a copy of `data` before making any modifications, see `updateAttributes` above.